macosx/Torrent.mm

```diff
@@ -2068,20 +2068,25 @@ - (BOOL)shouldShowEta
 
 - (NSString*)etaString
 {
-    NSInteger eta;
-    BOOL fromIdle;
-    //don't check for both, since if there's a regular ETA, the torrent isn't idle so it's meaningless
-    if (self.fStat->eta != TR_ETA_NOT_AVAIL && self.fStat->eta != TR_ETA_UNKNOWN)
+    BOOL unknown = NO;
+    time_t eta = self.fStat->eta;
+    // if there's a regular ETA, the torrent isn't idle
+    BOOL fromIdle = NO;
```

Comment:
I suggest to simplify all this.

As far as I see, `unknown` means early return, so:
```objc++
if (eta > INT_MAX || eta == TR_ETA_NOT_AVAIL || eta >= kETAIdleDisplaySec)
{
    return NSLocalizedString(@"remaining time unknown", "Torrent -> eta string");
}

if (eta == TR_ETA_UNKNOWN)
{
        eta = self.fStat->etaIdle;
        fromIdle = YES;
}
```

Maybe i'm missing something here?

Reply:
Yes, you're missing something:
- if fStat->eta is TR_ETA_NOT_AVAIL, we check for fStat->etaIdle
- kETAIdleDisplaySec is to be compared with fStat->etaIdle (not fStat->eta)

So we can't adopt your simplification, we need to keep the way I did. I also had a solution with `goto`, but it would have been frown upon.

Reply:
If `eta == TR_ETA_NOT_AVAIL` you set `unknown = YES` and then early return on `unknown`.
`self.fStat->etaIdle` is not used in this case.
What am I missing here?

Reply:
The second point:
- kETAIdleDisplaySec is to be compared with fStat->etaIdle (not fStat->eta)

Reply:
You're also missing the first point:
- if fStat->eta is TR_ETA_NOT_AVAIL, we check for fStat->etaIdle

My PR isn't changing the existing logic (which is already correct), it's just adding a cap for INT_MAX.

Reply:
@nevack I've now adopted a code change to avoid declaring `unknown`, if that makes it more readable for you?